docs/reference/components/sources/exec.cue

```diff
@@ -0,0 +1,237 @@
+			"x86_64-unknown-linux-musl":      true
+		}
+		requirements: []
+		warnings: []
+		notices: []
+	}
+
+	installation: {
+		platform_name: null
+	}
+
+	configuration: {
+		mode: {
+			description: "The type of exec mechanism."
+			required:    true
+			type: string: {
+				enum: {
+					scheduled: "Scheduled exec mechanism."
+					streaming: "Streaming exec mechanism."
+				}
+				syntax: "literal"
+			}
+		}
+		command: {
+			required:    true
+			description: "The command to be run."
+			type: string: {
+				examples: ["echo", "./myscript.sh"]
+				syntax: "literal"
+			}
+		}
+		arguments: {
+			common:      false
+			description: "Array of any arguments to pass to the command."
+			required:    false
+			type: array: {
+				default: null
+				items: type: [string]: {
+					examples: ["Hello World!", "-la"]
+					syntax: "literal"
+				}
+			}
+		}
+		current_dir: {
+			common:      false
+			required:    false
+			description: "The directory from within which to run the command."
+			warnings: []
+			type: string: {
+				default: null
+				syntax:  "literal"
+			}
+		}
+		include_stderr: {
+			common:      false
+			description: "Include the output of stderr when generating events."
+			required:    false
+			type: bool: default: false
+		}
+		event_per_line: {
+			common:      false
+			description: "Determine if events should be generated per line."
+			required:    false
+			type: bool: default: true
+		}
+		maximum_buffer_size: {
+			common:      false
+			description: "The maximum buffer size allowed before a log event will be generated."
+			required:    false
+			type: uint: {
+				default: 1000000
+				unit:    "bytes"
+			}
+		}
+		exec_interval_secs: {
+			common:        false
+			description:   "The interval in seconds between scheduled command runs. The command will be killed if it takes longer than exec_interval_secs to run."
+			relevant_when: "mode = `scheduled`"
+			required:      false
+			type: uint: {
+				default: 60
+				unit:    "seconds"
+			}
+		}
+		respawn_on_exit: {
+			common:        false
+			description:   "Determine if a streaming command should be restarted if it exits."
+			relevant_when: "mode = `streaming`"
+			required:      false
+			type: bool: default: true
+		}
+		respawn_interval_secs: {
+			common:        false
+			description:   "The interval in seconds between restarting streaming commands if needed."
+			relevant_when: "mode = `streaming`"
+			required:      false
+			warnings: []
+			type: uint: {
+				default: 60
+				unit:    "seconds"
+			}
+		}
+	}
+
+	output: logs: line: {
+		description: "An individual event from exec."
+		fields: {
+			data_stream: {
+				description: "The data stream from which the event originated."
+				required:    true
+				type: string: {
+					examples: ["stdout", "stderr"]
+					syntax: "literal"
+				}
+			}
+			exec_duration_millis: {
+				common:      false
+				description: "The duration in milliseconds a scheduled command took to complete."
```

Comment:
The `exec_duration_millis` is sent on an event along with `exit_status` when the command finishes (this applies to both a scheduled command and a streaming command if it exits). This event is sent after the command finishes and so unlike the previous events it will not have `data_stream` set (as stdout/stderr is not relevant to the finishing of the command).